### DISCUSSION TITLE
Should we set the maximum length or text to 860px and if so how should this work?

### DISCUSSION THREAD

**Discussion by AmnestyAM** [amnestywebsite/humanity-theme#106] [Enhancement Requests]
When the issue https://github.com/amnestywebsite/amnesty-wp-theme/issues/2708 was addressed this applied a max 860px on paragraphs, this was released by accident and reverted in https://github.com/amnestywebsite/humanity-theme/issues/99  . 

This change made the text easier to read which is good for accessibility, but left some strange empty spaces from a design perspective, so the change was reverted. 

https://www.amnesty.org/en/latest/research/2023/10/repression-trade/

White sections
![image](https://github.com/amnestywebsite/humanity-theme/assets/37809410/60619ab9-b96d-486a-b818-e6b219c93e3a)

Grey Sections
![image](https://github.com/amnestywebsite/humanity-theme/assets/37809410/b9aecb6d-1091-4506-b055-1400322bffe9)

sections in tabs block
![image](https://github.com/amnestywebsite/humanity-theme/assets/37809410/6df0aab7-3a00-4b5b-825d-a9eb11ef37c2)

Other text blocks such as paragraphs, headers, block, quotes etc
![image](https://github.com/amnestywebsite/humanity-theme/assets/37809410/48afa906-dc34-4ba4-a4b5-eba68e5bd06a)

We need to consider the companion plugin and the country pages that uses a different layout size, our chosen solution if possible should use the same logic as posts (with and without side bars) and pages, and we can hopefully can reduce the differences between templates here too. 

https://www.amnesty.org/en/location/asia-and-the-pacific/south-asia/afghanistan/
![image](https://github.com/amnestywebsite/humanity-theme/assets/37809410/b450c415-9dd0-4647-babd-18a1d7bf8d61)

```
.container--small {
    max-width: 1113px; 
}
```

on pages with a side bar the change looks fine as there is very little difference between the 860px limit and the previous limit.
https://www.amnesty.org/en/latest/news/2024/03/eu-artificial-intelligence-rulebook-fails-to-stop-proliferation-of-abusive-technologies/

```
.article.has-sidebar {
    max-width: 900px;
}
```

some things we will think about

We don't want to replicate the issue we had with the full width template which was to reduce the content "breaking out", when we had this feature it was hard for a user to align a page in a visual appealing way as the user need to either add 3 columns or add padding options in the section block. We removed these options which made it easier to create and publish content, we would like to avoid forcing users to do this again. 
https://github.com/amnestywebsite/amnesty-wp-theme/issues/2353 
![image](https://github.com/amnestywebsite/humanity-theme/assets/37809410/c0e0368b-fbec-4172-9570-a891a4f4c50c)

- Consider adding an option to section block to constrain width
- Consider asking users to use 3 columns
- Consider adding the restricted style when the txt is wider than a certain width, think about how we would mange page titles in the page meta area if this was done 
- Consider applying a constrained text to all content if the full width page option is applied 
- Consider making it easy for the CMS user 
- Consider reducing differences between templates, ie country pages should be the same if possible

![image](https://github.com/amnestywebsite/humanity-theme/assets/37809410/58756da1-ac9e-421e-aecf-7c2e904510ca)

**Comment by carolinejcourtney** [comment]
Commenting for traction, but also a potential suggestion...

Could this be something that we resolve with block patterns that use spacers within columns to create padding? The way I see it there should be three layout options for text, and each of them should meet the <860px requirement.

- Three-column layout, with only spacers in the left and right columns. Middle column contains text. I think there should be something we can do with the column block settings to make it so that the middle column is always less than 860px. (This should probably be the default option for sections of text?)
- Two-column layout, 50/50% split. Text on the left, image or video on the right. This is what we tend to use for Issue pages and Press release Plus pages on amnesty.org. 
- Left-aligned text on its own like the example you provided with the Overviews on the Country page.  

Also worth adding that whatever rules we apply to paragraphs should also apply to headings and blockquote, since these are often placed together and should follow the same alignment on the page.

**Reply by AmnestyAM** [reply]
Thanks for adding to the conversation here @carolinejcourtney yeah that is a way it could be fixed, and I think it would work, but would require the user to add a pattern, like in the way before we required a user to select a template. I really want something that requires nothing of the user, it might not be possible but that's what I want.

I want it so users can can open the page and just type. 

That is true for pages with a side bar, or pages when the side bar disabled, but not true when set to maximize content. 

![image](https://github.com/amnestywebsite/humanity-theme/assets/37809410/19e37a81-1aaa-482b-978d-d4ff1d214536)

with this change it might be a little less problematic https://github.com/amnestywebsite/amnesty-wp-theme/issues/2595 as we get a little extra padding on the left and right.

I have a feeling maximize content should be use by us only, and when selected we might need to apply some design changes like the max width being set on text, but we could probably set it every where anyway as all the other users will not be creating content that would break that principal/rule. 

We may need to add a page in the CMS with the setting enabled and add all the blocks to it and see what looks broken. and a rule like If grater than 860 then adding padding for the excess space ie if 900 then 20 left 860 for the text and 20 right...  I need to add more comment here but I am out of time for now, so need to hit reply.
